src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -381,8 +385,15 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
         } OT_TOOL_PACKED_END data;
 
         aMessage.Read(offset, sizeof(Tlv), &data.tlv);
-        aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
-        mLocal.Set(data.tlv);
+
+        if ((type != Tlv::kActiveTimestamp) ||
+            (data.tlv.GetType() != Tlv::kChannel && data.tlv.GetType() != Tlv::kMeshLocalPrefix &&
+             data.tlv.GetType() != Tlv::kPanId && data.tlv.GetType() != Tlv::kNetworkMasterKey))
```

Comment:
We should not be selectively applying TLVs here.  If an Active Dataset is received from a Thread device and the Active Dataset affects connectivity, those changes must be delayed.  See Thread 1.1 Section 8.7.4.8 for the expected behavior.